### PR TITLE
[M] Updated the gettext and msgfmt Gradle tasks

### PR DESCRIPTION
**Pull request by Ceiu**
- Added two parameters to allow specifying the input and output directories for the msgfmt Gradle task
- Added additional status output to the gettext and msgfmt tasks
- Minor refactors to the gettext task
- Updated the pom generator task to generate a pom.xml that no
  longer uses the gettest Maven plugin, and instead calls back
  into Gradle to generate source files

### CODE REVIEW

**Comment by Ceiu** [comment]
For reviewers:
This should work with any means of invoking this task -- either directly, or indirectly through Gradle dependency chains. Feel free to throw any weird combinations of things at it to try to make it use the wrong input or output directory.
If you happen to know Gradle's API well and see something that doesn't follow conventions or see a better way of doing it, please call it out.

**Review by nikosmoum — COMMENTED** [review]
Can we also get maven using the gradle msgfmt task and adding the generated .java files to the resources to compile?

File: buildSrc/src/main/groovy/Gettext.groovy [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- *  Copyright (c) 2009 - 2019 Red Hat, Inc.
+ *  Copyright (c) 2009 - ${YEAR} Red Hat, Inc.
```
Comment (nikosmoum) [inline buildSrc/src/main/groovy/Gettext.groovy:2]: 2023

File: buildSrc/src/main/groovy/Msgfmt.groovy [hunk 2]
```diff
@@ -13,72 +13,107 @@
...
-                                    "${po_file.canonicalPath}"
+                List msgfmt_args = [
+                    "--java2", "--source",
+                    "--resource", extension.resource,
+                    "-d", "${build_dir.canonicalPath}",
+                    "-l", locale,
+                    "${po_file.canonicalPath}"
                 ]
 
+                // println("Executing command: msgfmt ${msgfmt_args.join(" ")}")
```
[the remark this answers is not in the document]
Comment (Ceiu) [reply buildSrc/src/main/groovy/Msgfmt.groovy:107]: Much better. All of those `println`s from the areas I've touched have been replaced with proper logger statements.

**Review by nikosmoum — APPROVED** [review]

**Pull request merged**